src/backend/distributed/planner/merge_planner.c

```diff
@@ -1302,7 +1355,7 @@ ExtractMergeSourceRangeTableEntry(Query *query)
 	List *fromList = query->jointree->fromlist;
 
 	/* We should have only one RTE(MergeStmt->sourceRelation) in the from-list */
-	if (list_length(fromList) != 1)
+	if (list_length(fromList) != 1 && !joinSourceOk)
```

Comment:
Please remove the `&& !joinSourceOk `

Ideally this snippet of code should have been a simple` Assert(list_length(fromList) ==1)` but optimized builds will not fire it, so put it as an ERROR, unless Postgres community changes some architecture, we should never hit this.

Reply:
We can't write a negative-test case for this, to improve code coverage, may be just change it to `Assert` ?

Reply:
I kept the error, but also expanded the comment.

Let's be on the safe side and prefer throwing error instead of crash, int case there is some bug etc?